docs/controller.md

```diff
@@ -0,0 +1,108 @@
+## Features
+- Users can send pipelines containing actions to execute. These actions are basically shell commands to execute. These pipelines are `yaml` files.
+- Users can track there actions by getting the logs from the Agent, the states of the job : `TODO`, `PENDING`, `DOING`, `COMPLETED`, `ERROR`. Refer to the sections [actions/states](#States).
+- The controller makes sure that each actions are executed in the right order (by design) and doesn't execute the next action if the previous one has failed.
+## Pipeline
+A pipeline is a set of actions that are executed in sequence. It is represented as a `yaml` file (please, refer to the [structure](#structure) section for the reference of each sections of this file). A pipeline fails if one of its actions fail. If none of its actions fail then the pipeline succeeds.
+### Structure
+#### `actions`
+A pipeline is made up of one or more `actions`, which run sequentially. These actions declare a set of commands to run which implies to also declare a support to run on (for now this environment will be a container).
+#### `actions.<action_id>`
+`<action_id>` stands for the action identifier. Thanks to this identifier you will be able to get the action details through the API provided by the controller.
+
+There can be multiple actions in one pipeline but the `action id` must be unique.
+
+**Usage example** 
+```yaml
+actions:
+  postinstall:
+  ...
+```
+Here `postinstall` is the identifier of your action.
+#### `actions.<action_id>.configuration`
+The part where you will declare the environment on which the action is going to run. 
+>[!Note]
+> We chose a `configuration` section in our manifest because maybe in the future we will add a `variable` section to specify environment variables. 
+#### `actions.<action_id>.configuration.container`
+Here, you will declare the address of the container you want to run your action on.
+>[!Note]
+> Maybe in the future we are going to provide virtual machine support to run your actions, that is the reason why we named this part `container` and not for instance `os`.
+
+**Example :**
+```yaml
+actions:
+  postinstall:
+    configuration:
+      container: debian:latest
+```
+#### `actions.<action_id>.commands`
+`command` is a **list** of shell commands that will be executed during the action.
+**Example**
+```yaml
+actions:
+  postinstall:
+    configuration:
+      container: debian:latest
+    commands:
+      - apt update
+      - apt install mfa-postinstall
+```
+### Outputs
+The pipeline needs to inform the user on the state of the actions, therefore it needs to provide outputs. Outputs aim to describe each actions state to get an insight on what is going on in your pipeline. An output has an **header** that must have one of the following value : `TODO`, `PENDING`, `DOING`, `COMPLETED` and `ERROR`.
```

Comment:
The pipeline does not inform the user, the controller does.

I'm not sure what the format of an `output` is here. If that's a pipeline status, it should be described through a HTTP endpoint.

Reply:
Outputs are HTTP responses. You can find their specification here : https://github.com/dev-sys-do/sealci/pull/7

Reply:
If they're HTTP responses, they rename this section accordingly.

Reply:
Done, we renamed this section